compiler/rustc_error_codes/src/error_codes/E0311.md

```diff
@@ -0,0 +1,44 @@
+This error occurs when there is an unsatisfied outlives bound involving an
+elided region on a generic type parameter or associated type.
+
+Erroneous code example:
+
+```compile_fail,E0311
+fn no_restriction<T>(x: &()) -> &() {
+    with_restriction::<T>(x)
+}
+
+fn with_restriction<'a, T: 'a>(x: &'a ()) -> &'a () {
+    x
+}
+```
+
+Why doesn't this code compile? It helps to look at the lifetime bounds that are
+automatically added by the compiler. For more details see the documentation for
+[lifetime elision]( https://doc.rust-lang.org/reference/lifetime-elision.html).
+
+The compiler elides the lifetime of `x` and the return type to some arbitrary
+lifetime `'anon` in `no_restriction()`. The only information available to the
+compiler is that `'anon` is valid for the duration of the function. When
+calling `with_restriction()`, the compiler requires the completely unrelated
+type parameter `T` to outlive `'anon` because of the `T: 'a bound` in
+`with_restriction()`. This causes an error because `T` is not required to
+outlive `'anon` in `no_restriction()`.
+
+If `no_restriction()` were to use `&T` instead of `&()` as an argument, the
+compiler would have added an implied bound [implied
+bound](https://rust-lang.github.io/rfcs/2089-implied-bounds.html), causing this
```

Comment:
```suggestion
compiler would have added an
[implied bound](https://rust-lang.github.io/rfcs/2089-implied-bounds.html), causing this
```
oh wow, we really don't have a user facing doc for implied bounds xx that sucks

the rfc is about implied trait bounds. implied bounds already existed before that and are stable, unlike this rfc. We should really have a page in the reference for that

Reply:
opened https://github.com/rust-lang/reference/pull/1261 so maybe we should wait for that to land so that you can point there

Reply:
Thanks for opening the PR to add the reference material.  I added some comments on it and will hold this PR until that one lands.